cpp/oneapi/dal/backend/primitives/super_accumulator/int128.hpp

```diff
@@ -0,0 +1,224 @@
+/*******************************************************************************
+* Copyright 2021 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#pragma once
+
+#include <cstdint>
+#include <type_traits>
+
+#include "oneapi/dal/backend/primitives/common.hpp"
+
+namespace oneapi::dal::backend::primitives {
+
+union duality64 {
+    double floatingpoint;
+    std::uint64_t integer;
+    constexpr duality64(const std::uint64_t& val) : integer(val) {}
+    constexpr duality64(const double& val) : floatingpoint(val) {}
+};
+
+template <typename Impl>
+class int128_base {
+public:
+    using derived_t = Impl;
+
+    int128_base() = default;
+
+    std::int64_t& get_high_half() {
```

Comment:
set?

Reply:
Not set actually cause it simply provides access to the reference. It can be set via this function or it can be get in the same way